docs/contributing.rst

```diff
@@ -2,12 +2,653 @@
+    While pull requests are supposed to be based on changes that are ready to be tested,
+    reviewed, and eventually be merged to repositories; there is an exception to this:
+    Draft Pull Requests. We encourage Draft PRs if you just want to just start a
+    conversation about your code that isn’t in any state to be judged, and get feedback
+    as well as some guidance from others. Please read this GitHub `blog
+    <https://github.blog/2019-02-14-introducing-draft-pull-requests/>`_ about draft PRs.
+
+Please refer to Github's `Creating a pull request
+<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`_
+guide for the instructions. We are especially avoiding to detail these instructions
+here as there are manly referrals to GitHub's graphical user interface, which might be
+changed in the future.
+
+.. note::
+    Below are significant things about pull requests that can be very helpful throughout
+    the entire contribution process (i.e. review and merge) process when performed:
+
+3.7.1. Link PR to Issue
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Don't forget to link your PR to the issue you have been working on! This will help not
+only keep track of issues easier but also get them closed automatically once your PR is
+merged. You can refer to GitHub's `Linking a pull request to an issue
+<https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue>`_
+for this.
+
+3.7.2. Review your changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before you finalize opening the actual PR, it is a good practice to review the changes
+that you’ve made. You should be able to review all of the changes that will go into
+this PR just before you press the `Create pull request` button.
+
+If there is any changes you want to make in the PR, you can delay creating the PR
+and push new commits, revert existing changes, etc. You can then create the PR.
+
+3.7.3. Common Elements of Most Pull Requests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Despite the fact that pull requests can differ regarding their purposes (e.g.
+correction of a single typo in only one file could make a PR), most of the PRs may
+consist of code changes that should, most of the time, include some other elements
+with it. Having all of such elements addressed in a PR could make the review and
+merge process a lot easier. Assuming such code changes, these are what might
+accompany them:
+
+3.7.3.1. Unit tests
+~~~~~~~~~~~~~~~~~~~
+
+Virtually all new Uxarray code needs to include unit tests of it implemented.
+The UXarray project makes use of diverse technologies for unit testing as
+follows:
+
+* All the unit tests of every single Python module (i.e. `.py` file) should be
+  implemented as a separate test script under the :code:`\test` folder of the
+  repository's root directory.
+
+* The `pytest <https://docs.pytest.org/en/stable/contents.html>`_ testing framework
```

Comment:
This might be ignorance on my part, but why, if we're writing tests with unittest, do we run them with pytest? Why don't we just run them with unittest?

Also, depending on the answer to the above question, this might be a good place to mention the [practical differences](https://docs.pytest.org/en/7.1.x/how-to/usage.html#other-ways-of-calling-pytest) between running `pytest` directly vs `python3 -m pytest`

Reply:
A few benefits coming from using `pytest` are listed [here](https://docs.pytest.org/en/7.1.x/how-to/unittest.html#how-to-use-unittest-based-tests-with-pytest). I linked to that document in the part of the guide where we discuss `unittest` can be used instead of `pytest` as follows:

"a number of benefits/conveniences coming from using ``pytest`` can be
seen `<URL>` though"

I added a new command (`python -m pytest test/<test_script_name>.py`) and mentioned the benefit of using `python -m ...` instead of `pytest` as follows:

"the latter of which will also add the current directory to :code:`sys.path`"